internal/statemachine/state_machine.go

```diff
@@ -244,6 +245,28 @@ func (stateMachine *StateMachine) postProcessGadgetYaml() error {
 				}
 			}
 
+			if structure.Role == gadget.SystemBoot || structure.Label == gadget.SystemBoot {
+				// handle special syntax of rootfs:/<file path> in
+				// structure content. This is needed to allow images
+				// such as raspberry pi to source their kernel and
+				// initrd from the staged rootfs later in the build
+				// process.
+				relativeRootfsPath, err := filepathRel(
```

Comment:
hm, trying to grasp the usage of `filepath.Rel()` - what is the reason for doing that? Does this mean that snapd has some sanity-checking to make sure the path is relative or something, not allowing absolute paths?

This also makes me thinking: should we actually somehow double check that the resulting path (after substituting `rootfs:` to the rootfs) actually ends up accessing a path from the chroot's rootfs instead of some random place on the host system? Or does snapd already do that? Or maybe the usage of `Rel()` somehow helps here? I would expect not, right? Since in theory one could create a malicious gadget that could copy sensitive files into the image without the user knowing.